### PR TITLE
[Bugfix] Fix Crashing When Loading Modules With Batchnorm Stats

### PR DESCRIPTION
**Pull request by alex-jw-brooks**
Currently, the `AutoWeightsLoader` crashes if it encounters batch normalization stats in the model's safetensors. This is because the bn stats aren't part of the `named_parameters`, but they are part of the state dict, likely due to the reasons mentioned on the PyTorch forums [here](https://discuss.pytorch.org/t/batch-norm-parameters-not-included-in-model-parameters/10265).

This PR adds loading the bn stats for the the torch module `_BatchNorm`'s subclasses, as well as some tests that currently throw ValueErrors.

(This fix is needed for the upcoming granite speech models from our group. Currently, there aren't many models in vLLM that use batchnorm except for phi4mm audio, which disables it by default, which is why it probably hasn't popped up before).

CC @njhill @tlrmchlsmth

### CODE REVIEW

**Review by DarkLight1337 — APPROVED** [review]
Nice catch, thanks for fixing!
